### PR TITLE
Add Avatar Preview In Account Settings #513

### LINKED ISSUE #513: Add Avatar Preview In Account Settings
**Actual Behaviour**

Only drop down to select avatar is present

**Expected Behaviour**

User should be able to preview avatar

**Steps to reproduce it**

Settings -> Account

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #513

Changes:

- Added preview avatar functionality
- Upload avatar
- View different types of avatar available: Upload, Gravatar, Default

**Latest Link**: URL: [enchanting-soda.surge.sh](https://enchanting-soda.surge.sh)

**Updated surge link**: [pr-513-1-fossasia-susi-accounts.surge.sh](https://pr-513-1-fossasia-susi-accounts.surge.sh)

Old link: https://pr-513-fossasia-susi-accounts.surge.sh

Screenshots for the change:

screen shot 2018-09-21 at 3 16 54 am
screen shot 2018-09-21 at 3 17 02 am
screen shot 2018-09-21 at 3 17 08 am
screen shot 2018-09-21 at 3 17 13 am

### CODE REVIEW

**Comment by fragm3** [comment]
Travis error: https://github.com/fossasia/accounts.susi.ai/pull/511

**Comment by fragm3** [comment]
@akshatnitd @anshumanv @DNS-404 please review

File: src/components/Settings/Settings.react.js [hunk 1]
```diff
@@ -685,7 +687,124 @@ class Settings extends Component {
...
+          className="setting-avatar"
+          src={defaultAvatar}
+        />
+      );
+    } else {
+      avatarRender = (
+        <img
+          alt="Gravatar avatar"
+          className="setting-avatar"
+          src={gravatarAvatar}
```
Comment (akshatnitd) [inline src/components/Settings/Settings.react.js]: It should not be a constant image, it should display the Gravatar image of the user.

**Comment by fragm3** [comment]
@akshatnitd looking into this bug, good catch. Not sure why its happening, its just an input field.

File: src/components/Settings/Settings.react.js [hunk 2]
```diff
@@ -685,7 +687,124 @@ class Settings extends Component {
...
+      });
+    };
+    reader.readAsDataURL(file);
+  };
+
+  removeAvatarImage = () => {
+    this.setState({
+      file: '',
+      isAvatarAdded: false,
+      imagePreviewUrl: '',
```
Comment (akshatnitd) [inline src/components/Settings/Settings.react.js]: Reset `avatarSrc` while removing Avatar Image, this may be the reason for the bug.

Comment (fragm3) [reply src/components/Settings/Settings.react.js]: Tried it, not working. Will look into this today, appreciate your help!

Comment (fragm3) [reply src/components/Settings/Settings.react.js]: Resolved the error, turns out the input field was not reseting the value of target on uploading the same image.

**Comment by fragm3** [comment]
Updated link: [pr-513-1-fossasia-susi-accounts.surge.sh](https://pr-513-1-fossasia-susi-accounts.surge.sh)

Solved the issue. Please review @akshatnitd @Akshat-Jain @anshumanv

**Comment by anshumanv** [comment]
Sorry for the delay, can you please lookup why travis is failing?

**Comment by anshumanv** [comment]
Deployments will fail if travis fails, so not sure, let's wait for otherss

**Comment by fragm3** [comment]
@anshumanv on running chat susi, I get the following:
screenshot 2018-10-07 at 1 25 29 pm

This PR has complexity 22, can we let this pass? There must be some configuration to set this up.

**Comment by mariobehling** [comment]
Without travis passing the project will not build on Github pages.

File: src/components/Settings/Settings.react.js [hunk 3]
```diff
@@ -844,22 +970,7 @@ class Settings extends Component {
...
-                    />
-                    {this.state.uploadingAvatar ? (
-                      <CircularProgress color="#ffffff" size={32} />
-                    ) : (
-                      'Upload Avatar'
-                    )}
-                  </label>
-                </form>
-              )}
+              {avatarRender}
```
Comment (akshatnitd) [inline src/components/Settings/Settings.react.js]: Use `renderAvatar()` here.

File: src/components/Settings/Settings.react.js [hunk 4]
```diff
@@ -679,13 +681,135 @@ class Settings extends Component {
...
+    this.setState({
+      file: '',
+      isAvatarAdded: false,
+      imagePreviewUrl: '',
+      avatarSrc: '',
+    });
+  };
+
   render() {
+    let avatarRender = null;
```
Comment (akshatnitd) [inline src/components/Settings/Settings.react.js]: Move this logic to a function named `renderAvatar()` and then return the JSX from that function.

**Comment by fragm3** [comment]
@akshatnitd Thanks for the help. Please review @mariobehling @Akshat-Jain @aashutoshrathi @akshatnitd @DNS-404

**Comment by fragm3** [comment]
@akshatnitd Isn't it better to keep it this way only? If the user wants to remove the photo he will click on the cross, just saying.

**Comment by fragm3** [comment]
@akshatnitd Fixed, URL: [enchanting-soda.surge.sh](https://enchanting-soda.surge.sh)

**Pull request merged**